addons/website_sale/models/website.py

```diff
@@ -463,9 +463,8 @@ def _get_current_fiscal_position_id(self, partner_sudo):
         # If the current user is the website public user, the fiscal position
         # is computed according to geolocation.
         if request.website.partner_id.id == partner_sudo.id:
-            country_code = request.geoip.get('country_code')
-            if country_code:
-                country_id = self.env['res.country'].search([('code', '=', country_code)], limit=1).id
+            if request.geoip.country_code:
```

Comment:
may be use the `self._get_geoip_country_code` method (test if request exists)

Reply:
`self._get_geoip_country_code` is a method to safeguard the usage of `request`, here `request` is used in the line above so it is safe to use `request.geoip.country_code` directly